src/classes/TDTM_DMLgt200_TEST.cls

```diff
@@ -0,0 +1,465 @@
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @date 2017
+ * @group TDTM
+ * @group-content ../../ApexDocContent/TDTM.htm
+ * @description Test class to validate behavior changes related to TDTM Static Flag handling
+ *
+***/
+@isTest
+private class TDTM_DMLgt200_TEST {
+
+    /*********************************************************************************************************
+    * @description if you only want to run one test in this class, fill in its name here.
+    * if you want to run all tests, then use '*'
+    */
+    private static string strTestOnly = '*';
+
+    /* @description The number of records to insert for these tests. Inserting more than 200 records at once
+    * will cause triggers to be called twice - once for the first group of 200 and then again for the remainder.
+    */
+    private static Integer insertCnt = 210  ;
+
+    /*********************************************************************************************************
+    * @description Test creating 200+ Contacts with Affiliations
+    *   NOTE: THIS WILL FAIL IN AN ORG WITHOUT THE STATIC FLAG CHANGES
+    */
+    @isTest
+    static void test_contacts_with_affiliations() {
+        if (strTestOnly != '*' && strTestOnly != 'test_contacts_with_affiliations') return;
+
+        npe5__Affiliations_Settings__c affiliationsSettingsForTests = UTIL_CustomSettingsFacade.getAffiliationsSettingsForTests(
+                new npe5__Affiliations_Settings__c(npe5__Automatic_Affiliation_Creation_Turned_On__c = true));
+
+        Id orgRTId = [SELECT Id FROM RecordType WHERE SobjectType = 'Account' AND DeveloperName = 'Organization' AND IsActive = True LIMIT 1].Id;
```

Comment:
our tests cannot assume they have any account record types.  We tend to not specify the record type when creating test data.  And yes, if the org had the HH Account record type incorrectly set as default, the test might fail.  but that is probably a good thing, since that is an incorrect setup that will lead to problems in the org.